cmd/goal/clerk.go

```diff
@@ -155,6 +156,7 @@ func init() {
 	simulateCmd.Flags().StringVarP(&outFilename, "result-out", "o", "", "Filename for writing simulation result")
 	simulateCmd.Flags().BoolVar(&simulateAllowEmptySignatures, "allow-empty-signatures", false, "Allow transactions without signatures to be simulated as if they had correct signatures")
 	simulateCmd.Flags().BoolVar(&simulateAllowMoreLogging, "allow-more-logging", false, "Lift the limits on log opcode during simulation")
+	simulateCmd.Flags().Uint64Var(&simulateExtraBudget, "extra-app-budget", 0, "Apply extra budget during simulation")
```

Comment:
`allow-extra-budget` or `allow-extra-opcodes` or `allow-extra-opcode-budget`.

Reply:
As I consider it more, I see you are trying to let people specify the exact amount more, so the "allow" terminology feels funny, since it implies a boolean.

I guess we have two decisions to make a) Is it it valuable to give users such precise control of the extra budget? b) If so, perhaps `--allow-extra-budget` can be a useful alias for a nice default, like `--extra-opcode-budget=100000`

I guess the default would be `SimulateMaxExtraBudget`?

Reply:
My thinking is, a) it would be interesting to provide users the exact control of extra budget, and b) I agree the aliasing would be helpful, so see [review comments response 1](https://github.com/algorand/go-algorand/pull/5354/commits/be8d6cbb14665189220c24ffe34676f138cff9ea), where I add the `--allow-extra-budget` to maximize `simulateExtraBudget`.

Reply:
i agree but with the word opcode along with the word budget every time

Reply:
see [extra-app-budget -> extra-opcode-budget, overrides ptr to instance ui…](https://github.com/algorand/go-algorand/pull/5354/commits/d305f968408260845418af128fa4d5de54ddc1e2)